src/libs/DebugUtils.ts

```diff
@@ -551,6 +612,164 @@ function validateReportActionJSON(json: string) {
+    }
+
+    if (report.isPinned) {
+        return 'debug.reasonVisibleInLHN.pinnedByUser';
+    }
+
+    const reportIsSettled = report.statusNum === CONST.REPORT.STATUS_NUM.REIMBURSED;
+
+    if (ReportUtils.isExpenseRequest(report) && !reportIsSettled) {
+        return 'debug.reasonVisibleInLHN.isNonReimbursedIOU';
+    }
+
+    if (report.errorFields?.addWorkspaceRoom) {
+        return 'debug.reasonVisibleInLHN.hasAddWorkspaceRoomErrors';
+    }
+
+    if (isInFocusMode && ReportUtils.isUnread(report) && ReportUtils.getReportNotificationPreference(report) !== CONST.REPORT.NOTIFICATION_PREFERENCE.MUTE) {
+        return 'debug.reasonVisibleInLHN.isUnread';
+    }
+
+    if (isInDefaultMode && ReportUtils.isArchivedRoom(report, ReportUtils.getReportNameValuePairs(report?.reportID))) {
+        return 'debug.reasonVisibleInLHN.isArchived';
+    }
+
+    if (ReportUtils.isSelfDM(report)) {
+        return 'debug.reasonVisibleInLHN.isSelfDM';
+    }
+
+    return 'debug.reasonVisibleInLHN.isFocused';
+}
+
+/**
+ * Gets the reason that is causing the GBR to show up in LHN row
+ *
+ * @param report
+ * @param parentReportAction
+ * @returns translation key or null
+ */
+function getReasonForShowingGreenDotInLHNRow(report: OnyxEntry<Report>, parentReportAction?: ReportAction): TranslationPaths | null {
+    if (!report) {
+        return null;
+    }
+
+    if (ReportUtils.isJoinRequestInAdminRoom(report)) {
+        return 'debug.reasonGBR.hasJoinRequest';
+    }
+
+    if (ReportUtils.isUnreadWithMention(report)) {
+        return 'debug.reasonGBR.isUnreadWithMention';
+    }
+
+    if (ReportUtils.isWaitingForAssigneeToCompleteAction(report, parentReportAction)) {
+        return 'debug.reasonGBR.isWaitingForAssigneeToCompleteAction';
+    }
+
+    // Has a child report that is awaiting action (e.g. approve, pay, add bank account) from current user
+    if (report.hasOutstandingChildRequest) {
+        return 'debug.reasonGBR.hasChildReportAwaitingAction';
+    }
+
+    const invoiceRoomReportActions = ReportActionsUtils.getAllReportActions(report.reportID);
+    const hasMissinginvoiceBankAccountInInvoiceRoom =
+        ReportUtils.isInvoiceRoom(report) &&
+        Object.values(invoiceRoomReportActions).some(
+            (reportAction) =>
+                reportAction.actionName === CONST.REPORT.ACTIONS.TYPE.REPORT_PREVIEW && reportAction.childReportID && ReportUtils.hasMissingInvoiceBankAccount(reportAction.childReportID),
+        );
+    if (ReportUtils.hasMissingInvoiceBankAccount(report.reportID) || hasMissinginvoiceBankAccountInInvoiceRoom) {
+        return 'debug.reasonGBR.hasMissingInvoiceBankAccount';
+    }
+
+    return null;
+}
+
+/**
+ * Gets the report action that is causing the GBR to show up in LHN
+ *
+ * @param report
+ * @param reportActions
+ * @returns report action
+ */
+function getGBRReportAction(report: OnyxEntry<Report>, reportActions: OnyxEntry<ReportActions>): OnyxEntry<ReportAction> {
```

Comment:
```suggestion
/**
 * Gets the report action that is causing the GBR to show up in LHN.
 */
function getGBRReportAction(report: OnyxEntry<Report>, reportActions: OnyxEntry<ReportActions>): OnyxEntry<ReportAction> {
```

Could you also add a comment in the function about where you based this logic?